Failed to build with node v12.1.0

When I initialize a fresh project and try to add keyboard-layout the build with node-gyp fails.
I use node v12.1.0, yarn und 64bit Linux

Here's the output

```
yarn add electron-spellchecker
yarn add v1.13.0
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/3] ⢀ @nornagon/cld
[2/3] ⢀ @nornagon/spellchecker
error /home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@12.1.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [
gyp info spawn args   '/home/pk/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pk/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/pk/.node-gyp/12.1.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/pk/.node-gyp/12.1.0',
gyp info spawn args   '-Dnode_gyp_dir=/home/pk/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/pk/.node-gyp/12.1.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Verzeichnis „/home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout/build“ wird betreten
  CXX(target) Release/obj.target/keyboard-layout-manager/src/keyboard-layout-manager-linux.o
In file included from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
../../nan/nan.h: In Funktion »void Nan::AsyncQueueWorker(Nan::AsyncWorker*)«:
../../nan/nan.h:2232:62: Warnung: Umwandlung zwischen inkompatiblen Funktionstypen von »void (*)(uv_work_t*)« {aka »void (*)(uv_work_s*)«} nach »uv_after_work_cb« {aka »void (*)(uv_work_s*, int)«} [-Wcast-function-type]
     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
                                                              ^
../src/keyboard-layout-manager-linux.cc: In statischer Elementfunktion »static void KeyboardLayoutManager::Init(v8::Local<v8::Object>, v8::Local<v8::Object>)«:
../src/keyboard-layout-manager-linux.cc:19:78: Fehler: keine passende Funktion für Aufruf von »v8::FunctionTemplate::GetFunction()«
   module->Set(Nan::New("exports").ToLocalChecked(), newTemplate->GetFunction());
                                                                              ^
In file included from /home/pk/.node-gyp/12.1.0/include/node/node.h:63,
                 from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
/home/pk/.node-gyp/12.1.0/include/node/v8.h:5947:46: Anmerkung: candidate: »v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)«
   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
                                              ^~~~~~~~~~~
/home/pk/.node-gyp/12.1.0/include/node/v8.h:5947:46: Anmerkung:   Kandidat erwartet 1 Argument, 0 angegeben
In file included from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
../src/keyboard-layout-manager-linux.cc: Im globalen Gültigkeitsbereich:
/home/pk/.node-gyp/12.1.0/include/node/node.h:556:43: Warnung: Umwandlung zwischen inkompatiblen Funktionstypen von »void (*)(v8::Local<v8::Object>, v8::Local<v8::Object>)« nach »node::addon_register_func« {aka »void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)«} [-Wcast-function-type]
       (node::addon_register_func) (regfunc),                          \
                                           ^
/home/pk/.node-gyp/12.1.0/include/node/node.h:590:3: Anmerkung: bei Substitution des Makros »NODE_MODULE_X«
   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
   ^~~~~~~~~~~~~
../src/keyboard-layout-manager-linux.cc:22:1: Anmerkung: bei Substitution des Makros »NODE_MODULE«
 NODE_MODULE(keyboard_layout_manager, KeyboardLayoutManager::Init)
 ^~~~~~~~~~~
../src/keyboard-layout-manager-linux.cc: In statischer Elementfunktion »static Nan::NAN_METHOD_RETURN_TYPE KeyboardLayoutManager::GetCurrentKeymap(Nan::NAN_METHOD_ARGS_TYPE)«:
../src/keyboard-layout-manager-linux.cc:195:45: Warnung: »bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)« ist veraltet: Use maybe version [-Wdeprecated-declarations]
         entry->Set(unmodifiedKey, unmodified);
                                             ^
In file included from /home/pk/.node-gyp/12.1.0/include/node/v8-internal.h:14,
                 from /home/pk/.node-gyp/12.1.0/include/node/v8.h:25,
                 from /home/pk/.node-gyp/12.1.0/include/node/node.h:63,
                 from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
/home/pk/.node-gyp/12.1.0/include/node/v8.h:3359:26: Anmerkung: hier deklariert
                     bool Set(Local<Value> key, Local<Value> value));
                          ^~~
/home/pk/.node-gyp/12.1.0/include/node/v8config.h:322:3: Anmerkung: in Definition des Makros »V8_DEPRECATE_SOON«
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/keyboard-layout-manager-linux.cc:196:43: Warnung: »bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)« ist veraltet: Use maybe version [-Wdeprecated-declarations]
         entry->Set(withShiftKey, withShift);
                                           ^
In file included from /home/pk/.node-gyp/12.1.0/include/node/v8-internal.h:14,
                 from /home/pk/.node-gyp/12.1.0/include/node/v8.h:25,
                 from /home/pk/.node-gyp/12.1.0/include/node/node.h:63,
                 from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
/home/pk/.node-gyp/12.1.0/include/node/v8.h:3359:26: Anmerkung: hier deklariert
                     bool Set(Local<Value> key, Local<Value> value));
                          ^~~
/home/pk/.node-gyp/12.1.0/include/node/v8config.h:322:3: Anmerkung: in Definition des Makros »V8_DEPRECATE_SOON«
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../src/keyboard-layout-manager-linux.cc:197:39: Warnung: »bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)« ist veraltet: Use maybe version [-Wdeprecated-declarations]
         result->Set(dom3CodeKey, entry);
                                       ^
In file included from /home/pk/.node-gyp/12.1.0/include/node/v8-internal.h:14,
                 from /home/pk/.node-gyp/12.1.0/include/node/v8.h:25,
                 from /home/pk/.node-gyp/12.1.0/include/node/node.h:63,
                 from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
/home/pk/.node-gyp/12.1.0/include/node/v8.h:3359:26: Anmerkung: hier deklariert
                     bool Set(Local<Value> key, Local<Value> value));
                          ^~~
/home/pk/.node-gyp/12.1.0/include/node/v8config.h:322:3: Anmerkung: in Definition des Makros »V8_DEPRECATE_SOON«
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
In file included from /home/pk/.node-gyp/12.1.0/include/node/node.h:63,
                 from ../../nan/nan.h:53,
                 from ../src/keyboard-layout-manager.h:4,
                 from ../src/keyboard-layout-manager-linux.cc:1:
/home/pk/.node-gyp/12.1.0/include/node/v8.h: In Instanziierung von »void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]«:
/home/pk/.node-gyp/12.1.0/include/node/node_object_wrap.h:84:78:   von hier erfordert
/home/pk/.node-gyp/12.1.0/include/node/v8.h:9817:16: Warnung: Umwandlung zwischen inkompatiblen Funktionstypen von »v8::WeakCallbackInfo<node::ObjectWrap>::Callback« {aka »void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)«} nach »Callback« {aka »void (*)(const v8::WeakCallbackInfo<void>&)«} [-Wcast-function-type]
                reinterpret_cast<Callback>(callback), type);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/pk/.node-gyp/12.1.0/include/node/v8.h: In Instanziierung von »void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]«:
../../nan/nan_object_wrap.h:65:61:   von hier erfordert
/home/pk/.node-gyp/12.1.0/include/node/v8.h:9817:16: Warnung: Umwandlung zwischen inkompatiblen Funktionstypen von »v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback« {aka »void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)«} nach »Callback« {aka »void (*)(const v8::WeakCallbackInfo<void>&)«} [-Wcast-function-type]
make: *** [keyboard-layout-manager.target.mk:109: Release/obj.target/keyboard-layout-manager/src/keyboard-layout-manager-linux.o] Fehler 1
make: Verzeichnis „/home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout/build“ wird verlassen
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/pk/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:256:12)
gyp ERR! System Linux 5.0.9-2-MANJARO
gyp ERR! command "/home/pk/.nvm/versions/node/v12.1.0/bin/node" "/home/pk/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/pk/Downloads/spellcheckertest/node_modules/keyboard-layout
gyp ERR! node -v v12.1.0
gyp ERR! node-gyp -v v3.8.0
```

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,4 @@ node_modules/
 npm-debug.log
 *.swp
 build
+.vscode

--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
   ],
   "dependencies": {
     "event-kit": "^2.0.0",
-    "nan": "^2.10.0"
+    "nan": "^2.13.2"
   },
   "devDependencies": {
     "jasmine-focused": "^1.0.4"

--- a/src/keyboard-layout-manager-linux.cc
+++ b/src/keyboard-layout-manager-linux.cc
@@ -16,7 +16,9 @@ void KeyboardLayoutManager::Init(v8::Local<v8::Object> exports, v8::Local<v8::Ob
   Nan::SetMethod(proto, "getCurrentKeyboardLanguage", KeyboardLayoutManager::GetCurrentKeyboardLayout); // NB:  Intentionally mapped to same stub
   Nan::SetMethod(proto, "getInstalledKeyboardLanguages", KeyboardLayoutManager::GetInstalledKeyboardLanguages);
   Nan::SetMethod(proto, "getCurrentKeymap", KeyboardLayoutManager::GetCurrentKeymap);
-  module->Set(Nan::New("exports").ToLocalChecked(), newTemplate->GetFunction());
+
+  Nan::Set(module, Nan::New("exports").ToLocalChecked(),
+    Nan::GetFunction(newTemplate).ToLocalChecked());
 }
 
 NODE_MODULE(keyboard_layout_manager, KeyboardLayoutManager::Init)
@@ -192,9 +194,9 @@ NAN_METHOD(KeyboardLayoutManager::GetCurrentKeymap) {
 
       if (unmodified->IsString() || withShift->IsString()) {
         v8::Local<v8::Object> entry = Nan::New<v8::Object>();
-        entry->Set(unmodifiedKey, unmodified);
-        entry->Set(withShiftKey, withShift);
-        result->Set(dom3CodeKey, entry);
+        Nan::Set(entry, unmodifiedKey, unmodified);
+        Nan::Set(entry, withShiftKey, withShift);
+        Nan::Set(result, dom3CodeKey, entry);
       }
     }
   }

--- a/src/keyboard-layout-manager-mac.mm
+++ b/src/keyboard-layout-manager-mac.mm
@@ -22,7 +22,8 @@ void KeyboardLayoutManager::Init(Local<Object> exports, Local<Object> module) {
   Nan::SetMethod(proto, "getInstalledKeyboardLanguages", KeyboardLayoutManager::GetInstalledKeyboardLanguages);
   Nan::SetMethod(proto, "getCurrentKeymap", KeyboardLayoutManager::GetCurrentKeymap);
 
-  module->Set(Nan::New("exports").ToLocalChecked(), newTemplate->GetFunction());
+  Nan::Set(module, Nan::New("exports").ToLocalChecked(),
+    Nan::GetFunction(newTemplate).ToLocalChecked());
 }
 
 NODE_MODULE(keyboard_layout_manager, KeyboardLayoutManager::Init)
@@ -104,7 +105,7 @@ NAN_METHOD(KeyboardLayoutManager::GetInstalledKeyboardLanguages) {
     Local<Array> result = Nan::New<Array>(ret.size());
     for (size_t i = 0; i < ret.size(); ++i) {
        const std::string& lang = ret[i];
-       result->Set(i, Nan::New<String>(lang.data(), lang.size()).ToLocalChecked());
+       Nan::Set(result, i, Nan::New<String>(lang.data(), lang.size()).ToLocalChecked());
     }
 
     info.GetReturnValue().Set(result);
@@ -210,12 +211,12 @@ NAN_METHOD(KeyboardLayoutManager::GetCurrentKeymap) {
 
       if (unmodified->IsString() || withShift->IsString() || withAltGraph->IsString() || withAltGraphShift->IsString()) {
         Local<Object> entry = Nan::New<Object>();
-        entry->Set(unmodifiedKey, unmodified);
-        entry->Set(withShiftKey, withShift);
-        entry->Set(withAltGraphKey, withAltGraph);
-        entry->Set(withAltGraphShiftKey, withAltGraphShift);
+        Nan::Set(entry, unmodifiedKey, unmodified);
+        Nan::Set(entry, withShiftKey, withShift);
+        Nan::Set(entry, withAltGraphKey, withAltGraph);
+        Nan::Set(entry, withAltGraphShiftKey, withAltGraphShift);
 
-        result->Set(dom3CodeKey, entry);
+        Nan::Set(result, dom3CodeKey, entry);
       }
     }
   }

--- a/src/keyboard-layout-manager-windows.cc
+++ b/src/keyboard-layout-manager-windows.cc
@@ -42,6 +42,12 @@ HKL GetForegroundWindowHKL() {
 
 void KeyboardLayoutManager::Init(Local<Object> exports, Local<Object> module) {
   Nan::HandleScope scope;
+
+  // Once Nan supports Node v12, remove this.
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  // End remove after Nan supports Node v12
+
   Local<FunctionTemplate> newTemplate = Nan::New<FunctionTemplate>(KeyboardLayoutManager::New);
   newTemplate->SetClassName(Nan::New<String>("KeyboardLayoutManager").ToLocalChecked());
   newTemplate->InstanceTemplate()->SetInternalFieldCount(1);
@@ -52,8 +58,10 @@ void KeyboardLayoutManager::Init(Local<Object> exports, Local<Object> module) {
   Nan::SetMethod(proto, "getInstalledKeyboardLanguages", KeyboardLayoutManager::GetInstalledKeyboardLanguages);
   Nan::SetMethod(proto, "getCurrentKeymap", KeyboardLayoutManager::GetCurrentKeymap);
 
-  Nan::Set(module, Nan::New("exports").ToLocalChecked(),
-    Nan::GetFunction(newTemplate).ToLocalChecked());
+  // Note: Once NAN supports Node v12, change this to:
+  // Nan::Set(module, Nan::New("exports").ToLocalChecked(),
+  //   (Nan::GetFunction(newTemplate)).ToLocalChecked());
+  module->Set(Nan::New("exports").ToLocalChecked(), newTemplate->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(keyboard_layout_manager, KeyboardLayoutManager::Init)

--- a/src/keyboard-layout-manager-windows.cc
+++ b/src/keyboard-layout-manager-windows.cc
@@ -52,7 +52,8 @@ void KeyboardLayoutManager::Init(Local<Object> exports, Local<Object> module) {
   Nan::SetMethod(proto, "getInstalledKeyboardLanguages", KeyboardLayoutManager::GetInstalledKeyboardLanguages);
   Nan::SetMethod(proto, "getCurrentKeymap", KeyboardLayoutManager::GetCurrentKeymap);
 
-  module->Set(Nan::New("exports").ToLocalChecked(), newTemplate->GetFunction());
+  Nan::Set(module, Nan::New("exports").ToLocalChecked(),
+    Nan::GetFunction(newTemplate).ToLocalChecked());
 }
 
 NODE_MODULE(keyboard_layout_manager, KeyboardLayoutManager::Init)
@@ -119,7 +120,7 @@ NAN_METHOD(KeyboardLayoutManager::GetInstalledKeyboardLanguages) {
     wstr.assign(buf);
 
     std::string str = ToUTF8(wstr);
-    result->Set(i, Nan::New<String>(str.data(), str.size()).ToLocalChecked());
+    Nan::Set(result, i, Nan::New<String>(str.data(), str.size()).ToLocalChecked());
   }
 
   delete[] layouts;
@@ -209,12 +210,12 @@ NAN_METHOD(KeyboardLayoutManager::GetCurrentKeymap) {
 
       if (unmodified->IsString() || withShift->IsString() || withAltGraph->IsString() || withAltGraphShift->IsString()) {
         Local<Object> entry = Nan::New<Object>();
-        entry->Set(unmodifiedKey, unmodified);
-        entry->Set(withShiftKey, withShift);
-        entry->Set(withAltGraphKey, withAltGraph);
-        entry->Set(withAltGraphShiftKey, withAltGraphShift);
+        Nan::Set(entry, unmodifiedKey, unmodified);
+        Nan::Set(entry, withShiftKey, withShift);
+        Nan::Set(entry, withAltGraphKey, withAltGraph);
+        Nan::Set(entry, withAltGraphShiftKey, withAltGraphShift);
 
-        result->Set(dom3CodeKey, entry);
+        Nan::Set(result, dom3CodeKey, entry);
       }
     }
   }

